### ISSUE TITLE
Installation problem: There was an error with the webserver and the files to run offline are not present

### ISSUE THREAD

**Issue by captainwasabi** [micwallace/wallacepos#3]
OK, everything seems installed ok.  However when I go to the main page I get a popup that says "There was an error with the webserver and the files to run offline are not present."  

There is nothing in the webserver error.log.

![image](https://cloud.githubusercontent.com/assets/8583375/7413844/02a60500-ef1d-11e4-9c21-e421dae896c5.png)

Getting there.

**Comment by micwallace** [comment]
Hey,

Are you using ssl?
I may or may not have written some https:// links in the code. If so I'll have to update them to // but I suggest you use ssl on your install aswell!

Another thought is a 500 error.

You can check these using browser dev tools (firebug), look in the network section.

Cheers,
Michael


On 30 April 2015 11:41:24 PM AEST, captainwasabi <notifications@github.com> wrote:
>OK, everything seems installed ok.  However when I go to the main page
>I get a popup that says "There was an error with the webserver and the
>files to run offline are not present."  
>
>There is nothing in the webserver error.log.
>
>![image](https://cloud.githubusercontent.com/assets/8583375/7413844/02a60500-ef1d-11e4-9c21-e421dae896c5.png)
>
>
>Getting there.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/micwallace/wallacepos/issues/3

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by captainwasabi** [comment]
I'm not.  I'll set that up, np and see if that does it.  I was just installing to play around on a VM that only I can get to anyway so I didn't.

Nothing on the network section of the dev tools (I'm using chrome) but this was in the console:

Creating Application Cache with manifest http://pos/wpos.appcache
pos/:1 Application Cache Checking event
pos/:1 Application Cache Error event: Failed to parse manifest http://pos/wpos.appcache
 Consider using 'dppx' units, as in CSS 'dpi' means dots-per-CSS-inch, not dots-per-physical-inch, so does not correspond to the actual 'dpi' of a screen. In media query expression: only screen and (-webkit-min-device-pixel-ratio: 2), only screen and (min-resolution: 144dpi)
2 Consider using 'dppx' units, as in CSS 'dpi' means dots-per-CSS-inch, not dots-per-physical-inch, so does not correspond to the actual 'dpi' of a screen. In media query expression: only screen and (-webkit-min-device-pixel-ratio: 1.5), only screen and (min-resolution: 144dpi)
jquery-2.0.3.min.js:5 Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check http://xhr.spec.whatwg.org/.
http://pos/assets/js/jquery-2.0.3.min.js Failed to load resource: the server responded with a status of 404 (Not Found)x.support.cors.e.crossDomain.send @ jquery-2.0.3.min.js:5
core.js:463 Uncaught TypeError: "#loginbutton".prop is not a function

**Comment by micwallace** [comment]
Yeah i see a couple of 404 errors. Which means your apache is not configured correctly. I would be using an ip address rather than a self set domain name (pos) to rule out any dns issues.

http://pos/assets/js/jquery-2.0.3.min.js 
Change pos to an ip.

On 1 May 2015 8:20:05 AM AEST, captainwasabi <notifications@github.com> wrote:
>I'm not.  I'll set that up, np and see if that does it.  I was just
>installing to play around on a VM that only I can get to anyway so I
>didn't.
>
>Nothing on the network section of the dev tools (I'm using chrome) but
>this was in the console:
>
>Creating Application Cache with manifest http://pos/wpos.appcache
>pos/:1 Application Cache Checking event
>pos/:1 Application Cache Error event: Failed to parse manifest
>http://pos/wpos.appcache
>Consider using 'dppx' units, as in CSS 'dpi' means dots-per-CSS-inch,
>not dots-per-physical-inch, so does not correspond to the actual 'dpi'
>of a screen. In media query expression: only screen and
>(-webkit-min-device-pixel-ratio: 2), only screen and (min-resolution:
>144dpi)
>2 Consider using 'dppx' units, as in CSS 'dpi' means dots-per-CSS-inch,
>not dots-per-physical-inch, so does not correspond to the actual 'dpi'
>of a screen. In media query expression: only screen and
>(-webkit-min-device-pixel-ratio: 1.5), only screen and (min-resolution:
>144dpi)
>jquery-2.0.3.min.js:5 Synchronous XMLHttpRequest on the main thread is
>deprecated because of its detrimental effects to the end user's
>experience. For more help, check http://xhr.spec.whatwg.org/.
>http://pos/assets/js/jquery-2.0.3.min.js Failed to load resource: the
>server responded with a status of 404 (Not
>Found)x.support.cors.e.crossDomain.send @ jquery-2.0.3.min.js:5
>core.js:463 Uncaught TypeError: "#loginbutton".prop is not a function
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/micwallace/wallacepos/issues/3#issuecomment-97988455

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by captainwasabi** [comment]
no difference with just using the IP.  It'll take me a while to set up https (got too much else going on)

apache config:

<VirtualHost *:80>

        DocumentRoot /usr/share/wallacepos-master
        <Directory />
                Options FollowSymLinks
                AllowOverride None
        </Directory>

        <Directory /var/www/wallacepos-master>
                Options Indexes FollowSymLinks MultiViews
                AllowOverride None
                Order allow,deny
                allow from all
        </Directory>

        ErrorLog ${APACHE_LOG_DIR}/error.log

        # Possible values include: debug, info, notice, warn, error, crit,
        # alert, emerg.
        LogLevel warn

        CustomLog ${APACHE_LOG_DIR}/pos.access.log combined

</VirtualHost>

**Comment by micwallace** [comment]
Hey,

Here's my apache config for reference. I think you are missing some stuff.

This should go in /etc/apache2/sites-enabled/:

<VirtualHost *:443>
     DocumentRoot </var/www/yourrootdir> (wallacepos-master in your case)
     ServerName <your.server.name> (pos in your case)

     ErrorLog ${APACHE_LOG_DIR}/error.log
         CustomLog ${APACHE_LOG_DIR}/access.log combined

     SSLEngine on
         SSLCipherSuite !ADH:!DSS:!RC4:HIGH:+3DES:+RC4
         SSLProtocol all -SSLv2 -SSLv3
         SSLCertificateFile <certificate_location>
         SSLCertificateKeyFile <key_location>
         SSLCertificateChainFile <cert_chain_location>

     # WEBPROXT CONF
     ProxyRequests Off
     ProxyPreserveHost On
     <Proxy *>
             Order deny,allow
             Allow from all
     </Proxy>
     ProxyPass /socket.io/1/websocket/ 
ws://localhost:8080/socket.io/1/websocket/
     ProxyPassReverse /socket.io/1/websocket/ 
ws://localhost:8080/socket.io/1/websocket
     ProxyPass /socket.io/ http://localhost:8080/socket.io/
     ProxyPassReverse /socket.io/ http://localhost:8080/socket.io/
     <Location /socket.io>
             Order allow,deny
             Allow from all
     </Location>

</VirtualHost>

Replace <*> with your values.

Kind regards*,
*Michael*

*

**Comment by captainwasabi** [comment]
weird, it's not showing the <virtualHost> tags in my post, but they are there.  The Proxy stuff I put in /etc/apache2/apache2.conf but I moved them now.  I went ahead and set up ssl (who needs sleep?) but there was no difference.  The two files it says won't load: 

Application Cache Error event: Manifest fetch failed (9) https://192.168.2.53/wpos.appcache
Failed to load resource: the server responded with a status of 404 (Not Found)x.support.cors.e.crossDomain.send @ jquery-2.0.3.min.js:5

load fine if I load them directly:

https://192.168.2.53/wpos.appcache
https://192.168.2.53/assets/js/jquery-2.0.3.min.js

Both work and show the contents of the files...

**Comment by micwallace** [comment]
Hey,

Now that ssl is activated I think CORS is blocking the request when using IP.

Try to access the server via the domain/host name:
http://pos/

I think you also have to have this host specified in the ssl certificate.

Cheers,
Michael

PS: Don't stay up too late. Sleep is important :)

On 1 May 2015 2:09:09 PM AEST, captainwasabi <notifications@github.com> wrote:
>weird, it's not showing the <virtualHost> tags in my post, but they are
>there.  The Proxy stuff I put in /etc/apache2/apache2.conf but I moved
>them now.  I went ahead and set up ssl (who needs sleep?) but there was
>no difference.  The two files it says won't load: 
>
>Application Cache Error event: Manifest fetch failed (9)
>https://192.168.2.53/wpos.appcache
>Failed to load resource: the server responded with a status of 404 (Not
>Found)x.support.cors.e.crossDomain.send @ jquery-2.0.3.min.js:5
>
>load fine if I load them directly:
>
>https://192.168.2.53/wpos.appcache
>https://192.168.2.53/assets/js/jquery-2.0.3.min.js
>
>Both work and show the contents of the files...
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/micwallace/wallacepos/issues/3#issuecomment-98039279

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by captainwasabi** [comment]
http://pos
https://pos

both have the same result..

**Comment by micwallace** [comment]
Hey,

CORS:
http://en.wikipedia.org/wiki/Cross-origin_resource_sharing

The root cause of this is because you are accessing the page index.html 
via a different host/protocol than the page is loading the resources 
with. Either that or there's a certificate/webserver config problem.

This is fully weird as those assets are hard coded in index.html
Can you have a look in the firebug network section and see how the files 
are being access when you use https://pos/

Kind regards,
Michael*

*

**Comment by captainwasabi** [comment]
In the networking tab everything seems good except this:

https://pos/api/hello?_=1430481427853

In the /api directory there are just two files, server.js and wpos.php.


![image](https://cloud.githubusercontent.com/assets/8583375/7429930/a95fa73c-efd8-11e4-8d39-47038f29fa63.png)

**Comment by micwallace** [comment]
ah ok,

There's also a .htaccess file in that folder that redirects all requests 
to wpos.php.
Have you got mod_rewrite enabled?

Kind regards,
Michael*

*

**Comment by captainwasabi** [comment]
enabled mod_rewwrite,  Set the directory to allow overrides in the site config, restarted apache, no difference.

Current site config:

    <VirtualHost *:443>

        DocumentRoot /usr/share/wallacepos-master
        ServerName pos

         SSLEngine on
         SSLCipherSuite !ADH:!DSS:!RC4:HIGH:+3DES:+RC4
         SSLProtocol all -SSLv2 -SSLv3
         SSLCertificateFile /root/ssl/pos.crt
         SSLCertificateKeyFile /root/ssl/pos.key
         #SSLCertificateChainFile /root/ssl/

  <Directory /usr/share/wallacepos-master>
    AllowOverride all
  </Directory>


        ErrorLog ${APACHE_LOG_DIR}/error.log

        # Possible values include: debug, info, notice, warn, error, crit,
        # alert, emerg.
        LogLevel warn

        CustomLog ${APACHE_LOG_DIR}/pos.access.log combined

    ProxyRequests Off
    ProxyPreserveHost On
    <Proxy *>
        Order deny,allow
        Allow from all
    </Proxy>
    ProxyPass /socket.io/1/websocket/ ws://127.0.0.1:8080/socket.io/1/websocket/
    ProxyPassReverse /socket.io/1/websocket/ ws://127.0.0.1:8080/socket.io/1/websocket/
    ProxyPass /socket.io/ http://127.0.0.1:8080/socket.io/
    ProxyPassReverse /socket.io/ http://127.0.0.1:8080/socket.io/
    <Location /socket.io>
        Order allow,deny
        Allow from all
    </Location>


    </VirtualHost>

**Comment by micwallace** [comment]
Hey

Override directive needs to be in a Directory tag like this:

<Directory /usr/share/wallacepos-master>
     AllowOverride All
</Directory>

And I'm pretty sure it needs to be placed outside the VirtualHost tag.

**Comment by captainwasabi** [comment]
Oh I edited my comment,  It did make a difference.  It now says the device is not registered or something, then when I try to login as admin/admin it says it can't connect to the host. (Sorry, if you're just using email then you don't see the updates so I won't do that anymore.

**Comment by captainwasabi** [comment]
ok, I just killed my system somehow. :( I'm just going to close this.  I've got to reinstall a VM from the ground up. :(

**Comment by micwallace** [comment]
Hi Wasabi,

You were very close!
"connect to the host" error is because the node.js component (the feed server that provides realtime funtionality) is not running.

Run this and it should solve the issue:
nodejs /api/server.js

After you rebuild the vm, if your still running into problems, I'm happy to log onto your vm and have a look if you provide me access via ssh and https.

It's the least I can do for helping me write a proper install guide :)

Cheers,
Michael

PS: For interest, which country are you from? I'm in Sydney, Australia.

On 2 May 2015 6:41:25 AM AEST, captainwasabi <notifications@github.com> wrote:
>ok, I just killed my system somehow. :( I'm just going to close this. 
>I've got to reinstall a VM from the ground up. :(
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/micwallace/wallacepos/issues/3#issuecomment-98230982

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by captainwasabi** [comment]
I'm in the US.

It seems most of the problems have been resolved (sans new the issues I reported on github that are waiting for when you wake up and get to them ;)

Besides the improvements you made I think re-installing on a 14.04 VM solved a lot of issues because it ships with the correct apache and php versions.

However, not supporting multiple tax authorities per location might preclude me from using this.  Still evaluating.

Thanks!
